models/ShapePath/ShapePath.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+const uuid = require('uuid-v4');
+const Layer = require('../Layer');
+const Rect = require('../Rect');
+const Style = require('../Style');
+
+/**
+ *
+ */
+class ShapePath extends Layer {
+  /**
+   *
+   * @param {Object} args
+   * @param {String} args.name
+   * @param {integer} args.x
+   * @param {integer} args.y
+   * @param {integer} args.height
+   * @param {integer} args.width
+   * @param {Array.<CurvePoint>} args.points
+   * @param {Object} args.style Passed to {@link LayerStyle}
+   * @param {ShapePath.Model} json
+   */
+  constructor(args = {}, json) {
+    super(args, json);
+    if (!json) {
+      const id = args.id || uuid().toUpperCase();
+      Object.assign(this, ShapePath.Model, {
+        points: args.points || [],
+        do_objectID: id,
+        frame: new Rect({
+          x: args.x,
+          y: args.y,
+          width: args.width,
+          height: args.height,
+        }),
+        style: new Style(args.style),
+        name: args.name || id,
+      });
+    }
+  }
+}
+
+/**
+ * @mixes Layer.Model
+ * @property {boolean} edited
+ * @property {boolean} isClosed
+ * @property {integer} pointRadiusBehaviour
+ * @property {String[]} points
+ */
+ShapePath.Model = Object.assign({}, Layer.Model, {
+  _class: 'shapePath',
+  edited: true,
+  isClosed: false,
+  pointRadiusBehaviour: 1,
```

Comment:
For these int enums could we create a map like this? https://github.com/amzn/sketch-constructor/blob/master/models/Border/Border.js#L52

Reply:
Hi Danny, I like the idea. Do you know what the values actually stand for? I'm having a hard time figure out what `pointRadiusBehaviour` is

Reply:
I just did some quick testing (by creating a sketch file and unzipping the contents), and I could not get that value to change. 😕 So yea, we don't need a map for this for now because I don't know what it actually does! LGTM!

Reply:
@dbanksdesign Yeah, I actually did the same. It's kinda weird. Maybe something to keep in mind for moving forward.